lib/publiccloud/aws_client.pm

```diff
@@ -74,6 +74,10 @@ sub configure_podman {
           . " | podman login --username AWS --password-stdin $full_name_prefix");
 }
 
+sub destroy {
```

Comment:
all `*_client` classes ( aws_client / azure_client / gce_client ) has  `cleanup` method empty . And it does make sense because `*_clients` classes are not responsible for any resources , so I don't see why we actually need them . So instead of adding one more phantom method which no one ever call can you please drop "cleanup" method below , together with useless `$self->provider_client->cleanup();` calls in provider's children ?